### PR TITLE
AtenSortOp Folder

### PR DESCRIPTION
**Pull request by renxida**
A chunk off

https://github.com/llvm/torch-mlir/pull/2856
https://github.com/llvm/torch-mlir/pull/2860

### CODE REVIEW

File: lib/Dialect/Torch/IR/TorchOps.cpp [hunk 1]
```diff
@@ -1688,6 +1688,41 @@ void AtenSortIntOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
...
 
+
+//===----------------------------------------------------------------------===//
+// AtenSortOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult AtenSortOp::fold(FoldAdaptor adaptor,
+                               SmallVectorImpl<OpFoldResult> &results) {
+  auto operand = getSelf();
+  auto operandTy = dyn_cast_if_present<ValueTensorType>(operand.getType());
```
Comment (newling) [inline lib/Dialect/Torch/IR/TorchOps.cpp:1702]: When would operand.getType() be nullptr? I'm wondering why you're using dyn_cast_if_present and not dyn_cast.

File: lib/Dialect/Torch/IR/TorchOps.cpp [hunk 2]
```diff
@@ -1688,6 +1688,41 @@ void AtenSortIntOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
...
+  if (!operandTy || !operandTy.hasSizes())
+    return failure();
+  
+  auto iTTy = dyn_cast_if_present<ValueTensorType>(getResult(1).getType());
+  auto indicesTy = iTTy ? iTTy.toBuiltinTensor().clone(iTTy.getDtype()) : nullptr;
```
Comment (newling) [inline lib/Dialect/Torch/IR/TorchOps.cpp]: I think it'd be clearer if you check !iTTy and return failure before getting indicesTy. Are you trying to minimize the number of exit points from the method?

**Review by newling — COMMENTED** [review]
(names) : use full names for variables when possible operandType, indicesType, etc.

File: lib/Dialect/Torch/IR/TorchOps.cpp (hunk 1 above)
Comment (renxida) [reply lib/Dialect/Torch/IR/TorchOps.cpp]: good point, using dyn_cast for this and the other one

File: lib/Dialect/Torch/IR/TorchOps.cpp (hunk 2 above)
Comment (renxida) [reply lib/Dialect/Torch/IR/TorchOps.cpp]: was thinking there should be a single line i can read to figure out why this folder could fail, but then realized that for debugging, having different lines probably makes it a lot easier

**Comment by renxida** [comment]
running into a lot of trouble because of the llvm version bumps : /

File: lib/Dialect/Torch/IR/TorchOps.cpp [hunk 3]
```diff
@@ -1688,6 +1692,49 @@ void AtenSortIntOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
...
 }
 
+//===----------------------------------------------------------------------===//
+// AtenSortOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult AtenSortOp::fold(FoldAdaptor adaptor,
+                               SmallVectorImpl<OpFoldResult> &results) {
+  auto operand = getSelf();
+  auto operandType = dyn_cast<ValueTensorType>(operand.getType());
```
Comment (newling) [inline lib/Dialect/Torch/IR/TorchOps.cpp:1702]: I wasn't sure about asserting that the type is ValueTensorType, I asked a q here: https://discord.com/channels/636084430946959380/742573221882364009/1204378906925342750

Comment (renxida) [reply lib/Dialect/Torch/IR/TorchOps.cpp]: Fixed!

**Comment by renxida** [comment]
Got approval from rob on discord.

**Pull request merged**